src/esp/assets/ResourceManager.cpp

```diff
@@ -649,6 +684,113 @@ Magnum::Range3D ResourceManager::computeMeshBB(BaseMesh* meshDataGL) {
+  const std::vector<PTexMeshData::MeshData>& submeshes = ptexMeshData.meshes();
+
+  for (uint32_t iEntry = 0; iEntry < absTransforms.size(); ++iEntry) {
+    // convert std::vector<vec3f> to std::vector<Mn::Vector3>
+    std::vector<Mn::Vector3> pos;
+    uint32_t meshID = staticDrawableInfo_[iEntry].second;
+    for (auto& p : submeshes[meshID].vbo) {
+      pos.emplace_back(p);
+    }
+
+    // transform the vertex positions to the world space
+    Mn::MeshTools::transformPointsInPlace(absTransforms[iEntry], pos);
+
+    // locate the scene node which contains the current drawable
+    esp::gfx::Drawable& drawable = staticDrawableInfo_[iEntry].first.get();
+    scene::SceneNode& node = dynamic_cast<scene::SceneNode&>(drawable.object());
+
+    // set the absolute axis aligned bounding box
+    node.setAbsoluteAABB(Mn::Range3D{Mn::Math::minmax<Mn::Vector3>(pos)});
+  }
+}
+
+void ResourceManager::computeGeneralMeshAbsoluteAABBs() {
+  std::vector<Mn::Matrix4> absTransforms = computeAbsoluteTransformations();
+
+  CORRADE_ASSERT(absTransforms.size() == staticDrawableInfo_.size(),
+                 "ResourceManager::computeGeneralMeshAbsoluteAABBs: number of "
+                 "transforms does not match number of drawables.", );
+
+  for (uint32_t iEntry = 0; iEntry < absTransforms.size(); ++iEntry) {
+    esp::gfx::Drawable& drawable = staticDrawableInfo_[iEntry].first.get();
+    uint32_t meshID = staticDrawableInfo_[iEntry].second;
+
+    Corrade::Containers::Optional<Magnum::Trade::MeshData3D>& meshData =
+        meshes_[meshID]->getMeshData();
+    CORRADE_ASSERT(meshData,
+                   "ResourceManager::computeGeneralMeshAbsoluteAABBs: the "
+                   "empty mesh data", );
+
+    // a vector to store the min, max pos for the aabb of every position array
+    std::vector<Mn::Vector3> bbPos;
+
+    // transform the vertex positions to the world space, compute the aabb for
+    // each position array
+    for (uint32_t jArray = 0; jArray < (*meshData).positionArrayCount();
+         ++jArray) {
+      std::vector<Mn::Vector3>& pos = (*meshData).positions(jArray);
+      std::vector<Mn::Vector3> absPos =
+          Mn::MeshTools::transformPoints(absTransforms[iEntry], pos);
+
+      std::pair<Mn::Vector3, Mn::Vector3> bb =
+          Mn::Math::minmax<Mn::Vector3>(absPos);
+      bbPos.push_back(bb.first);
+      bbPos.push_back(bb.second);
+    }
+
+    // locate the scene node which contains the current drawable
+    scene::SceneNode& node = dynamic_cast<scene::SceneNode&>(drawable.object());
+
+    // set the absolute axis aligned bounding box
+    node.setAbsoluteAABB(Mn::Range3D{Mn::Math::minmax<Mn::Vector3>(bbPos)});
+
+  }  // iEntry
+}
+
+std::vector<Mn::Matrix4> ResourceManager::computeAbsoluteTransformations() {
+  // sanity check
+  if (staticDrawableInfo_.size() == 0) {
+    return std::vector<Mn::Matrix4>{};
+  }
+
+  // basic assumption is that all the drawables are in the same scene;
```

Comment:
I think this assumption is wrong.  The resource manager may have multiple meshes loaded.  i.e. like an MP3D mesh where you have both the color mesh and semantic mesh loaded.  That'll also be true for semantic gibson meshes.

Reply:
See my comments above in ResourceManager.cpp. The program will not collect drawables from the semantic mesh.

Reply:
Also see my description in the "Motivation and Context" section.